cmd/controller-manager/pkg/features/kube_features.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package features
+
+import (
+	"k8s.io/apimachinery/pkg/util/runtime"
+
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+)
+
+const (
+	// Every feature gate should add method here following this template:
+	//
+	// // owner: @username
+	// // alpha: v1.4
+	// MyFeature() bool
+
+	// owner @smarterclayton
+	// alpha: v1.16
+	// beta:  v1.19
+	//
+	// Enable legacy behavior to vary cluster functionality on the node-role.kubernetes.io labels. On by default (legacy), will be turned off in 1.18.
+	LegacyNodeRoleBehavior featuregate.Feature = "LegacyNodeRoleBehavior"
+
+	// owner @brendandburns
+	// alpha: v1.9
+	// beta:  v1.19
+	//
+	// Enable nodes to exclude themselves from service load balancers
+	ServiceNodeExclusion featuregate.Feature = "ServiceNodeExclusion"
+)
+
+func init() {
```

Comment:
Rather than use an init block, can you make this `func AddCurrentKubernetesSpecificFeatureGates(featureGates featuregate.MutableFeatureGate) error {`

Reply:
Moved the call from register here. Thanks

Reply:
Switched back to using init block because calling it here would fail the integration test by close feature gates early which blocked future features to be added.

Reply:
> Switched back to using init block because calling it here would fail the integration test by close feature gates early which blocked future features to be added.

Can we get to the bottom of this?  As a minimum, provide the register function here and move the init block to a different package so we can separate the two concepts.  As a potential consumer, I don't want to grabbing teh constants to auto-register into an opinionated feature gate instance.